subprojects/android-test/ui-testing-core-app/src/androidTest/kotlin/com/avito/android/ui/test/dialog/DialogsTest.kt

```diff
@@ -0,0 +1,40 @@
+package com.avito.android.ui.test.dialog
+
+import androidx.test.espresso.NoMatchingViewException
+import com.avito.android.test.app.core.screenRule
+import com.avito.android.ui.DialogsActivity
+import com.avito.android.ui.test.Screen
+import org.junit.Rule
+import org.junit.Test
+import ru.avito.util.assertThrows
+
+class DialogsTest {
+
+    @get:Rule
+    val rule = screenRule<DialogsActivity>()
+
+    @Test
+    fun screen_isDisplayed__no_dialog() {
+        rule.launchActivity(null)
+
+        Screen.dialogsScreen.content.checks.isDisplayed()
+    }
+
+    @Test
+    fun dialog_isDisplayed__alert_dialog() {
+        rule.launchActivity(
+            DialogsActivity.intent(openDialog = true)
+        )
+        Screen.alert.messageElement.checks.isDisplayed()
+    }
+
+    @Test
+    fun screen_isNotDisplayed__alert_dialog() {
+        rule.launchActivity(
+            DialogsActivity.intent(openDialog = true)
```

Comment:
I don't get why we expect the error there

Reply:
We check the default behavior of matchers here.

`dialogsScreen.content` has an interaction without `Matcher<Root>`.
Espresso selects automatically the best root. 
When it selects the dialog as the root, it won't find the view.

Reply:
Maybe rename test case to
`open dialog and content without Root - no matching view exception`
because 
`screen_isNotDisplayed__alert_dialog` is not readable notation for me

Reply:
Agree
Renamed to `matcher_with_default_root__no_matching_view_exception__dialog_is_open`